packages/opds-web-client/src/interfaces.ts

```diff
@@ -192,3 +192,9 @@ export interface BasicAuthMethod extends AuthMethod {
     password: string;
   };
 }
+
+type PickAndRequire<T, K extends keyof T> = { [P in K]-?: NonNullable<T[P]> };
```

Comment:
Can you add a comment about what this is doing? Looks like just making sure all the properties from K are present, and also not null?

Reply:
Yep will do. It allows you to pick which properties you want to force to be required: 
```js
type BookDataWithCopies = RequireKeys<BookData, "copies">;
```

Reply:
Actually, `PickAndRequire` was only really usable in the context of `RequireKeys`, so I just inlined that type into `RequireKeys`, which does what I explained above